### PR TITLE
[GEOS-8769] Add integration tests and documentation about the usage of App-Schema with Apache Solr

### PR DESCRIPTION
**Pull request by nmco**
Associated issue:
https://osgeo-org.atlassian.net/browse/GEOS-8769

### CODE REVIEW

File: doc/en/user/source/data/app-schema/solr-tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
+.. _solr_tutorial:
+
+Apache Solr Tutorial
+====================
+
+This tutorial demonstrates how to use App-Schema plugin with a Apache Solr data store. This tutorial will focus on the Apache Solr data store specificities is highly recommended to read the App-Schema documentation before.
```
[the remark this answers is not in the document]
Comment (nmco) [reply doc/en/user/source/data/app-schema/solr-tutorial.rst]: Fixed.

File: doc/en/user/source/data/app-schema/solr-tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+The use case for this tutorial will be to serve through App-Schema the information about some meteorological stations index in an Apache Solr core. Note that this use case is completely fictional and only used to demonstrate the Apache Solr and App-Schema integration.
+
+A station data is composed of some meta-information about the station, e.g. is name and position. The only extra \ different configuration we need to provide when using Apache Solr as a data source is the configuration of the data store itself.
```
[the remark this answers is not in the document]
Comment (nmco) [reply doc/en/user/source/data/app-schema/solr-tutorial.rst]: Fixed.

File: doc/en/user/source/data/app-schema/solr-tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          <srid>4326</srid>
+          <type>POINT</type>
+        </geometry>
+      </index>
+    </SolrDataStore>
+  </sourceDataStores>
+
+In this particular case the the ``location`` attribute contains a point geometry and will be the default geometry.
+
+Follows a the complete mappings file:
```
[the remark this answers is not in the document]
Comment (nmco) [reply doc/en/user/source/data/app-schema/solr-tutorial.rst]: Fixed.

File: src/extension/app-schema/app-schema-solr-test/README.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+The fixture file only needs to contain the URL of a valid Apache Solr instance core.
+, no tests where made with SolarCloud. The tests will take care of creating
+the necessary schemas and load the necessary data.
+
+Unfortunately is not possible to create using Solar API a new core in a
+reliable way. So before running this plugin integration tests a core needs to be
+manually created.
+
+Is worth notice that support for advanced geometries, like polygons, is necessary. Which
```
[the remark this answers is not in the document]
Comment (nmco) [reply src/extension/app-schema/app-schema-solr-test/README.rst]: Fixed.

File: src/extension/app-schema/app-schema-solr-test/README.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        ~/.geoserver/solr.properties
+
+The fixture file should look like this:
+
+.. code-block:: properties
+
+        solr_url=http://localhost:8983/solr/test_core
+
+The fixture file only needs to contain the URL of a valid Apache Solr instance core.
+, no tests where made with SolarCloud. The tests will take care of creating
```
[the remark this answers is not in the document]
Comment (nmco) [reply src/extension/app-schema/app-schema-solr-test/README.rst]: Fixed.

File: src/extension/app-schema/app-schema-solr-test/README.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        http://localhost:8983/solr/admin/cores?action=CREATE&name=test_core&instanceDir=test_core
+
+        {
+          "responseHeader":{
+            "status":0,
+            "QTime":782},
+          "core":"test_core"}
+
+Note that the integration tests will be able to reuse the same core for multiple runs as long
+the core managed schema is not manually edited.
```
[the remark this answers is not in the document]
Comment (nmco) [reply src/extension/app-schema/app-schema-solr-test/README.rst]: Fixed.

**Comment by nmco** [comment]
Thank you for the documentation review @bradh, much appreciated ! I applied your suggestions.

**Comment by bencaradocdavies** [comment]
Failing on the bogus CoverageViewTest. Needs a close-reopen once the build is fixed.

**Comment by bencaradocdavies** [comment]
`ComplexSolrTest` formatting fixed in https://github.com/geoserver/geoserver/commit/79490ce50f19d994a914fda60240245a7690d521 .

**Comment by bencaradocdavies** [comment]
I would add `-Dfmt.action=check` to the Jenkins geoserver-master-app-schema-online build but GeoServer Jenkins is down.

**Comment by bencaradocdavies** [comment]
I have added `-Dfmt.action=check` to the Jenkins geoserver-master-app-schema-online build.

**Pull request merged**